cpgvalidator/src/main/scala/io/shiftleft/cpgvalidator/CpgValidator.scala

```diff
@@ -0,0 +1,193 @@
+package io.shiftleft.cpgvalidator
+
+import io.shiftleft.codepropertygraph.Cpg
+import gremlin.scala.{Edge, Vertex}
+import io.shiftleft.cpgvalidator.FactConstructionClasses.{InFact, OutFact}
+import org.apache.logging.log4j.LogManager
+import org.apache.tinkerpop.gremlin.structure.Direction
+
+import scala.collection.JavaConverters._
+
+class CpgValidator(notEnhancedCpg: Cpg) {
+  private val errorRegistry = new ValidationErrorRegistry()
+
+  def validate(): Boolean = {
+    validateOutFacts()
+    validateInFacts()
+
+    errorRegistry.getErrorCount == 0
+  }
+
+  def logValidationErrors(): Unit = {
+    errorRegistry.logValidationErrors()
+  }
+
+  private def validateOutFacts(): Unit = {
```

Comment:
`validateOutFacts` and `validateInFacts` look pretty similar. Any chance we can reduce some code duplication here?

Reply:
Yes can be deduplicated. So far i had no time to do it.